event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java

```diff
@@ -63,9 +68,46 @@ public static DbCandidate toDbCandidate(CristinNviReport cristinNviReport) {
                    .basePoints(extractBasePoints(cristinNviReport))
                    .collaborationFactor(extractCollaborationFactor(cristinNviReport))
                    .internationalCollaboration(isInternationalCollaboration(cristinNviReport))
+                   .channelId(extractChannelId(cristinNviReport))
                    .build();
     }
 
+    private static URI extractChannelId(CristinNviReport cristinNviReport) {
+        var instance = toInstanceType(cristinNviReport.instanceType());
+        var referenceNode =
+            attempt(() -> dtoObjectMapper.readTree(cristinNviReport.reference().toString())).orElse(failure -> null);
+        if (nonNull(instance)) {
```

Comment:
Cristin nvi candidate can have other InstanceType than our nvi service allow. Thats we have to check that instanceType enum is not null